addons/controllers/secretgen-controller/secretgencontrollerconfig_controller.go

```diff
@@ -0,0 +1,238 @@
+		// patchCRD will be true if finalizer or ownerrefence is added
+		if patchCRD {
+			log.Info("Patching SecretGenControllerConfig")
+
+			if err := patchHelper.Patch(ctx, secretGenConfig.DeepCopy()); err != nil {
+				log.Error(err, "Error patching SecretGenControllerConfig")
+				retErr = err
+			}
+		}
+	}()
+
+	// If SecretGenControllerConfig is marked for deletion then delete the data value secret
+	if !secretGenConfig.GetDeletionTimestamp().IsZero() {
+		err := r.ReconcileSecretGenControllerConfigDelete(ctx, log, clusterClient, secretGenConfig, &patchCRD)
+		if err != nil {
+			log.Error(err, "Error reconciling SecretGenControllerConfig delete")
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	if err := r.ReconcileSecretGenControllerConfigNormal(ctx, log, cluster, clusterClient, secretGenConfig, &patchCRD); err != nil {
+		log.Error(err, "Error reconciling SecretGenControllerConfig to create data value secret")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// ReconcileSecretGenControllerConfigNormal reconciles SecretGenControllerConfig data values secret
+func (r *SecretGenControllerConfigReconciler) ReconcileSecretGenControllerConfigNormal(
+	ctx context.Context,
+	log logr.Logger,
+	cluster *clusterapiv1beta1.Cluster,
+	clusterClient client.Client,
+	secretGenConfig *addonsv1alpha1.SecretGenControllerConfig,
+	patchCRD *bool) (retErr error) {
+
+	// Add finalizer to addon secret
+	*patchCRD = util.AddFinalizerToCRD(log, constants.SecretGenControllerAddonName, secretGenConfig)
+
+	// add owner reference to addon secret
+	ownerReference := metav1.OwnerReference{
+		APIVersion:         clusterapiv1beta1.GroupVersion.String(),
+		Kind:               "Cluster",
+		Name:               cluster.Name,
+		UID:                cluster.UID,
+		Controller:         pointer.BoolPtr(true),
+		BlockOwnerDeletion: pointer.BoolPtr(true),
+	}
+
+	if !clusterapiutil.HasOwnerRef(secretGenConfig.OwnerReferences, ownerReference) {
+		log.Info("Adding owner reference to SecretGenControllerConfig")
+		secretGenConfig.OwnerReferences = clusterapiutil.EnsureOwnerRef(secretGenConfig.OwnerReferences, ownerReference)
+		*patchCRD = true
+	}
+
+	addonDataValuesSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      util.GenerateDataValueSecretNameFromAddonNames(secretGenConfig.Name, constants.SecretGenControllerAddonName),
+			Namespace: secretGenConfig.Namespace,
+		},
+	}
+
+	addonDataValuesSecretMutateFn := func() error {
+		addonDataValuesSecret.Type = corev1.SecretTypeOpaque
+		addonDataValuesSecret.Data = map[string][]byte{}
+
+		// marshall the yaml contents
+		secretGenConfigYaml := &SecretGenControllerConfigSpecYaml{Namespace: secretGenConfig.Spec.Namespace, CreateNamespace: secretGenConfig.Spec.CreateNamespace}
+
+		yamlBytes, err := yaml.Marshal(secretGenConfigYaml)
+		if err != nil {
+			return err
+		}
+
+		dataValueBytes := append([]byte(constants.TKGDataValueFormatString), yamlBytes...)
```

Comment:
Do we need YTT annotations for or case in the header or just key valued configuration values wwould be enough?